Python/pythonrun.c

```diff
@@ -249,6 +260,19 @@ PyRun_InteractiveOneObject(FILE *fp, PyObject *filename, PyCompilerFlags *flags)
 }
 
 int
+PyRun_InteractiveOneObject(FILE *fp, PyObject *filename, PyCompilerFlags *flags)
+{
+    int res;
+
+    res = PyRun_InteractiveOneObjectEx(fp, filename, flags);
+    if (res == -1) {
```

Comment:
Maybe we shouldn't call PyErr_Print() + flush_io() in the worst MemoryError case which *cleared* the error (PyErr_Clear()), since flush_io() is likely to raise a new exception.

What do you think of replacing "if (res == -1)" with "if (res == -1 && PyErr_Occurred())" ?

Reply:
I think there may be a confusion: the worst MemoryError case which cleared the error (PyErr_Clear()) occurs in PyRun_InteractiveLoopFlags(), but here we are in PyRun_InteractiveOneObject() and this function does not call PyRun_InteractiveLoopFlags().

The substitution you suggest makes sense, but the initial version of PyRun_InteractiveOneObject() did not call PyErr_Occurred() and I prefer not to change that. What is modified by the PR though, is that now when PyImport_AddModuleObject() returns NULL in PyRun_InteractiveOneObjectEx() then PyErr_Print() is called now while it is not in the initial version.

Reply:
Right, I was confused between the "One" and "Loop" flavors. You can leave it unchanged.